packages/fuels-accounts/src/provider.rs

```diff
@@ -2,6 +2,7 @@ use std::{collections::HashMap, fmt::Debug, io, net::SocketAddr};
 
 mod retry_util;
 mod retryable_client;
+#[cfg(feature = "fuel-core-lib")]
```

Comment:
If I understand correctly, the supported version will only be checked when `fuel-core-lib`is used. If `fuel-core-lib` is used we compile `fuel-core` ourselfs which means that we will use the correct versione from our `Cargo.toml`.

Would it be possible to hardcore the supported version to a file during the publish step of our CI ?

Reply:
> If I understand correctly, the supported version will only be checked when fuel-core-libis used. If fuel-core-lib is used we compile fuel-core ourselfs which means that we will use the correct versione from our Cargo.toml.

Yeah, this type of check seems less critical than needing to do something similar for when you're _not_ using the `fuel-core-lib`. e.g., when you're talking to an external (external to the SDK code) node.

What are the current blockers/difficulties in doing something like that?

> Would it be possible to hardcore the supported version to a file during the publish step of our CI ?

How would that look and what are the benefits of doing it? At first this sounds like having the `fuel-core` lib version pinned in our `Cargo.toml` 🤔 but I'm probably missing something.